go/chat/unfurl/settings.go

```diff
@@ -0,0 +1,116 @@
+package unfurl
+
+import (
+	"context"
+	"sort"
+
+	"github.com/keybase/client/go/chat/types"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/chat1"
+)
+
+const settingsModeName = "__unfurl_settings_mode"
+const settingsWhitelistName = "__unfurl_settings_whitelist"
+
+type modeRecord struct {
+	Mode chat1.UnfurlMode
+}
+
+type whitelistRecord struct {
+	Whitelist []string
+}
+
+type Settings struct {
+	utils.DebugLabeler
+
+	storage types.ConversationBackedStorage
+}
+
+func NewSettings(log logger.Logger, storage types.ConversationBackedStorage) *Settings {
+	return &Settings{
+		DebugLabeler: utils.NewDebugLabeler(log, "Settings", false),
+		storage:      storage,
+	}
+}
+
+func (s *Settings) Get(ctx context.Context) (res chat1.UnfurlSettings, err error) {
+	defer s.Trace(ctx, func() error { return err }, "Get")()
+
+	var mr modeRecord
+	found, err := s.storage.Get(ctx, settingsModeName, &mr)
+	if err != nil {
+		return res, err
+	}
+	if !found {
+		s.Debug(ctx, "Get: no mode setting found, using whitelisted")
+		mr.Mode = chat1.UnfurlMode_WHITELISTED
+	}
```

Comment:
abort whitelist fetch if mode isn't whitelisted?

Reply:
I wanted to always get the whitelist for rendering the settings screen, where I think the whitelist will still show if you are in always or never mode.

Reply:
ah makes sense